src/resolve/resolved-dns-question.c

```diff
@@ -418,6 +418,66 @@ int dns_question_new_reverse(DnsQuestion **ret, int family, const union in_addr_
         return 0;
 }
 
+int dns_question_new_service_type(
+                DnsQuestion **ret,
+                const char *service,
```

Comment:
this is used to generate the PTR query that point to the services, hence this will always have a NULL service param, hence why take that argument at all? drop this?